### PR TITLE
Busywork changes 5/26/2006

### PR DESCRIPTION
**Pull request by bcbrock**
## Description

This changeset only touches files in fabric/tools/busywork

This changeset includes the following major features:
1. Added the 'networkStatus' script that reports busywork network
   status, and can also be used as a watchdog. The counters/driver
   now supports this script.
2. Added the 'pprofClient' script that interfaces to the
   profiling server in the peer and can also be used as a
   watchdog. The counters/driver now supports this script.

Minor updates:

fabricLogger has better error reporting
userModeNetwork starts the profiling server by default now (it should not hurt anything to do this normally).
Added the CLI port to userModeNetwork : Note, I've never tested CLI with user-mode networks as there has never been a need for it.
Doc. updates
Varius minor fixes
## Motivation and Context

See above.
## How Has This Been Tested?

I've been testing all of this stuff over the last few days as I have been developing it, esp. to collect profiles of peer networks.

In the very near future I hope that a few busywork tests will become part of the CI testing. For the moment I don't have tests for busywork itself, I'll have to think about what that would mean.

No Go code was changed.
## Checklist:

- [x] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff)
- [x] Either no new documentation is required by this change, OR I added new documentation
- [?] Either no new tests are required by this change, OR I added new tests
- [] I have run [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports), [go vet](https://golang.org/cmd/vet/), and [golint](https://github.com/golang/lint). I have cleaned up all valid errors and warnings in code I have added or modified. These tools may generate false positives. Don't be worried about ignoring some errors or warnings. The goal is clean, consistent, and readable code.

Signed-off-by: Bishop Brock bcbrock@us.ibm.com

### CODE REVIEW

**Comment by christo4ferris** [comment]
no go code was harmed by this PR

File: tools/busywork/bin/README.md [hunk 1]
```diff
@@ -11,25 +12,37 @@ fabric networks and obtaining logs from containers. Some of the make targets
...
+  network agree on the contents of the blockchain.
 
 * [fabricLogger](fabricLogger) is a simple script that converts a blockchain
   into a text file listing of transaction IDs.
   
-* [userModeNetwork](userModeNetwork) is a script defines Hyperledger fabric
-  peer networks as simple networks of user-mode processes, that is, networks
-  that do not rely on Docker-compose or multiple physical nodes. The concept
-  of a user-mode network is discussed [below](#userModeNetwork).
+* [networkStatus](networkStatis) is a script that reports on the status of
```
Comment (ghaskins) [inline tools/busywork/bin/README.md]: Intentional typo? (networkStatis)

**Comment by ghaskins** [comment]
Hi Bishop,

This tooling looks awesome, thank you.  As I mentioned in the initial commit, I am not qualified to give a formal code review as I don't know Tcl.  However, given that it is completely decoupled from the project currently, I am inclined to not worry too much about it.

So, with those caveats:

LGTM (after the typo noted is fixed), at least w.r.t. impact/risk assessment.

Reviewed-by: Greg Haskins gregory.haskins@gmail.com

**Comment by bcbrock** [comment]
@ghaskins Thanks for finding that typo Greg! I went through and also fixed another broken link, and added some minor doc. and functional updates.

**Comment by srderson** [comment]
LGTM

**Pull request merged**